vars/mavenIntegrationTest.groovy

```diff
@@ -2,35 +2,37 @@
 import io.fabric8.Utils
 
 def call(body) {
```

Comment:
How about this 

```groovy
def call(body) {
    def utils = new Utils()

    if (utils.isDisabledITests()) {
        echo "WARNING: Integration tests DISABLED for these pipelines!"
        return
    }

    // evaluate the body block, and collect configuration into the object
    def config = [:]
    body.resolveStrategy = Closure.DELEGATE_FIRST
    body.delegate = config
    body()


    if (cmd.integrationTestCmd) {
      sh integrationTestCmd
      junitResults(body);
      return
    }


    def envName = config.environment
    def kubeNS = "-Dfabric8.environment=${envName}"

    if (envName) {
        try {
            def ns = utils.environmentNamespace(envName)
            if (ns) {
                kubeNS = "-Dnamespace.use.existing=${ns}"
                echo "Running the integration tests in namespace : ${ns}"
            }
        } catch (e) {
            echo "ERROR: failed to find the environment namespace for ${envName} due to ${e}"
            e.printStackTrace()
        }
    }

    def testCmd =  "mvn org.apache.maven.plugins:maven-failsafe-plugin:integration-test \
      -P openshift-it ${kubeNS} \
      -Dit.test=${config.itestPattern} \
      -DfailIfNoTests=${config.failIfNoTests}  \
      org.apache.maven.plugins:maven-failsafe-plugin:verify"
    sh testCmd
    junitResults(body);
}
```

Reply:
I haven't tested this but the idea is quite simple ... The `call` either makes the `cmd` to run or uses the `cmd` passed to it and it is quite clear IMHO

Reply:
this looks more readable +1